crates/krun/src/net.rs

```diff
@@ -0,0 +1,53 @@
+use std::{
+    fmt,
+    os::{
+        fd::{AsRawFd, IntoRawFd, OwnedFd},
+        unix::net::UnixStream,
+    },
+    path::Path,
+    process::Command,
+};
+
+use anyhow::{anyhow, Context, Result};
+use log::debug;
+use rustix::net::{socketpair, AddressFamily, SocketFlags, SocketType};
+use utils::env::find_in_path;
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum NetMode {
+    PASST = 0,
+    TSI,
+}
+
+pub fn connect_to_passt<P>(passt_socket_path: P) -> Result<UnixStream>
+where
+    P: AsRef<Path> + fmt::Debug,
+{
+    Ok(UnixStream::connect(passt_socket_path)?)
+}
+
+pub fn start_passt() -> Result<OwnedFd> {
+    let (parent_fd, child_fd) = socketpair(
+        AddressFamily::UNIX,
+        SocketType::STREAM,
+        SocketFlags::empty(),
+        None,
+    )?;
+
+    let passt_path = find_in_path("passt").context("Failed to look up `passt` in PATH")?;
+    let passt_path = passt_path.ok_or_else(|| anyhow!("`passt` executable not found in PATH"))?;
+
+    debug!(fd = child_fd.as_raw_fd(); "passing fd to passt");
+
+    // SAFETY: `child_fd` is an `OwnedFd` and consumed to prevent closing on drop.
+    // See https://doc.rust-lang.org/std/io/index.html#io-safety
+    let child = Command::new(passt_path)
```

Comment:
We'd have to explicitly kill the `Child`: https://doc.rust-lang.org/std/process/struct.Child.html#method.kill

Also, it seems like we're exiting with `-1` status code, perhaps because of `sommelier`?

Reply:
This is still a problem...

Reply:
This is a limitation of `libkrun` (https://github.com/containers/libkrun/issues/10). We need to implement a mechanism to relay signals to the main process in the VM and the exit code from it to the host. It's been open since a long time, but being a low priority issue we never got to implement it.